### PR TITLE
Add transaction op default

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## :loudspeaker: Type of change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description
`transaction.op` is `required` on Android.

## :bulb: Motivation and Context
closes: https://github.com/getsentry/sentry-react-native/issues/2682

## :green_heart: How did you test it?
tests and sample app

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps

- https://github.com/getsentry/sentry-react-native/issues/2693

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,9 @@
 
 - Add `lastEventId` method to the API ([#2675](https://github.com/getsentry/sentry-react-native/pull/2675))
 
+### Fix
+- `Sentry.startTransaction` doesn't require `op` ([#2691](https://github.com/getsentry/sentry-react-native/pull/2691))
```
Comment (marandaneto) [inline CHANGELOG.md:10]:
```suggestion
### Fix

- `Sentry.startTransaction` doesn't require `op` ([#2691](https://github.com/getsentry/sentry-react-native/pull/2691))
```

File: src/js/measurements.ts [hunk 2]
```diff
@@ -48,6 +48,10 @@ const _patchStartTransaction = (
     transactionContext: TransactionContext,
     customSamplingContext?: CustomSamplingContext
   ): Transaction {
+    if (!transactionContext.op) {
+      transactionContext.op = 'default';
+    }
```
Comment (marandaneto) [inline src/js/measurements.ts:53]: We'll need to do that for `startChild` as well.

File: src/js/measurements.ts [hunk 3]
```diff
@@ -48,10 +48,23 @@ const _patchStartTransaction = (
     transactionContext: TransactionContext,
     customSamplingContext?: CustomSamplingContext
   ): Transaction {
+    if (!transactionContext.op) {
+      transactionContext.op = 'default';
```
Comment (marandaneto) [inline src/js/measurements.ts:52]: M: Let's make a `const` for `default`, its being used in multiple laces.

Comment (marandaneto) [inline src/js/measurements.ts:52]: M: I'm wondering if this should live here, the file is about measurements, is the reason that we only patch the `startTransaction` and `startChild` here? Can we move this patching else where?

Comment (marandaneto) [reply src/js/measurements.ts:52]: Let's add a comment explaining why we have to patch and set the `op` as `default`.
Same for `startChild`.

**Comment by marandaneto** [comment]
Let's double check if `startIdleTransaction` has to be patched as well, if its an internal and not documented API, we dont need to do that since we always have an `op`.

**Comment by krystofwoldrich** [comment]
@marandaneto Checked. `startIdleTransaction` is not documented and it's only used by the SDKs.

File: src/js/measurements.ts (hunk 3 above)
Comment (krystofwoldrich) [reply src/js/measurements.ts:52]: I'll open a new issue for the refactoring.

**Review by marandaneto — APPROVED** [review]

**Pull request merged**